### PR TITLE
libc/chdir: replace heap alloc to path buffer to improve performance

### PR DESCRIPTION
**Pull request by anchao**
## Summary

libc/chdir: replace heap alloc to path buffer to improve performance

realloc path buffer to avoid alloc from realpath()

Signed-off-by: chao an <anchao@lixiang.com>

## Impact

N/A

## Testing

sim/nsh

### CODE REVIEW

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by nuttxpr** [comment]
[**\[Experimental Bot, please feedback here\]**](https://github.com/search?q=repo%3Aapache%2Fnuttx+13552&type=issues)



No, this PR does not fully meet the NuttX requirements.  While it provides a summary of *what* was changed, it lacks crucial details.  Here's why:

* **Insufficient Summary Detail:** The summary mentions performance improvement, but doesn't explain *why* the current implementation is slow or *how* the change improves performance (e.g., by how much?). It also lacks context on the functional part of the code being changed (beyond `libc/chdir`).

* **Missing Impact Assessment:**  "N/A" is insufficient.  Even if the changes seem minor, each impact area needs to be explicitly addressed with a "YES" or "NO" and a brief explanation.  For instance, even if there's no user-facing change, the impact on user should be explicitly stated as "NO - No changes to user interface or behavior."  There might be implications for memory usage which should be discussed.

* **Incomplete Testing Information:**  While `sim/nsh` is mentioned, it's not enough.  The PR needs to specify:
    * **Build Host Details:**  OS, CPU architecture, compiler version used for building.
    * **Target Details:**  Specific simulator configuration used (e.g., `sim:nsh`).
    * **Missing Logs:**  The "Testing logs before change" and "Testing logs after change" sections are empty.  These are crucial to demonstrate the improvement and that the functionality remains correct.  Showing timing differences would be particularly helpful here.  Even a simple "time ./nsh" before and after would be a big improvement.

* **Missing Issue References:** If this change relates to a reported issue, the PR should reference the relevant NuttX or NuttX Apps issue.

In short, the PR needs to provide substantially more detail in the Summary, Impact, and Testing sections to meet the NuttX requirements.

File: boards/arm/mps/mps2-an500/configs/knsh/defconfig [hunk 1]
```diff
@@ -61,7 +61,6 @@ CONFIG_READLINE_CMD_HISTORY=y
 CONFIG_RR_INTERVAL=200
 CONFIG_SCHED_HPWORK=y
 CONFIG_SCHED_HPWORKPRIORITY=192
-CONFIG_SPINLOCK=y
```
Comment (xiaoxiang781216) [inline boards/arm/mps/mps2-an500/configs/knsh/defconfig:64]: @crafcat7 please look at this problem.

Comment (anchao) [reply boards/arm/mps/mps2-an500/configs/knsh/defconfig:64]: So, Let's merge this commit first ? which does not conflict with the current feature

Comment (xiaoxiang781216) [reply boards/arm/mps/mps2-an500/configs/knsh/defconfig:64]: Ok.

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**